### ISSUE TITLE
Pen is not being saved as Template

### ISSUE THREAD

**Issue by reccanti** [CodePen/CodePen-Bug-Reporting-Only#61]
Hello!

I'm having some issues saving a pen as a template. When I check the "template?" checkbox, it stays checked while I'm on the page, but when I switch from that view, it doesn't appear in my list of templates. I've tried reloading the pages with caching disabled so I don't think that is the issue.

**Comment by mariemosley** [comment]
Also reported via Twitter (link includes video demo: https://twitter.com/shshaw/status/690213252358213632)

Thanks for opening the issue @reccanti!

**Comment by reccanti** [comment]
Has there been any progress on this issue? I haven't heard anything in a while.

**Comment by chriscoyier** [comment]
The trick is trying to find a way to replicate the bug. Shaw's video is a perfect replication. but when I try to do it...

Video: http://d.pr/v/17b0z/5u7IpGsL (works fine)

So we gotta figure of it's it's account-related, or some other weird quirk. Maybe try a video with the console open so we can see if some kind of error throws there?

**Comment by shshaw** [comment]
Is it possible that the issue only occurs on non-pro accounts? @reccanti, are you using pro or 'lite'?

**Comment by shshaw** [comment]
No console errors. http://cl.ly/0D3M0D1C421k

**Comment by reccanti** [comment]
I'm using a 'lite' account. I'll see if I can get a video up soon!

**Comment by mariemosley** [comment]
I was able to reproduce this exactly once, but didn't get a screencast for it. Here's what I was doing:

- Using my free CodePen account
- Using Chrome
- Gave the template this title: Hello, Template.

Perhaps the punctuation in the title has something to do with it?

Could not reproduce this at all on my PRO account, and couldn't get it to happen more than once on the free account.

**Comment by shshaw** [comment]
I don't think punctuation affects it; in my recent demo I am not using punctuation but it still does not save as template.

**Comment by yukulele** [comment]
If I edit "title" and "template?" then save, only "title" is saved.
request called (codepen.io/pen/save/) contain "template" in is parameters and returned json data with expected `true` value in both

**Comment by chriscoyier** [comment]
So we got EIGHT people around a table trying every possible thing to reproduce this and we just can't figure it out. Haha. It's the worst bug ever. 

We're gonna close this for now, since we haven't had any more reports for 27 days. Maybe it's fixed? We'll definitely reopen this (and fix it) if we can figure out a way to reproduce it.

**Comment by shshaw** [comment]
That's unfortunate. I can replicate it every time over here!

For reference:
- Non-pro account, authenticated through GitHub.
- Mac OS X 10.11.3
- Chrome Version 49.0.2623.87 (64-bit), in Incognito and regular.
- Firefox 44.0.2

Editor settings: 
![Editor settings](https://cloud.githubusercontent.com/assets/777155/13931092/bab4e14a-ef6f-11e5-9b3f-8bb21b31a438.png)

Oddly enough, I'm noticing that under Account Activity, it's saying that I 'forked' the new Pens, instead of creating a new pen. Not sure if that's typical.

![screen shot 2016-03-21 at 2 21 08 pm](https://cloud.githubusercontent.com/assets/777155/13931190/376cc752-ef70-11e5-8d4e-49977ba7d05b.png)

**Comment by yukulele** [comment]
Non-pro account, authenticated through GitHub too.
windows, firefox or chrome.

**Comment by chriscoyier** [comment]
@shshaw that **forking** this is a great lead. What the heck, right? Activity says you forked it, but it's not listed as a fork in your profile **nor** does it make it into your templates. That's a bug, whether or not it's related to this. (I logged in as you and tried it myself with your account).

When saving the Pen with the template checkbox checked, it does send the data across the wire correctly. The JSON contains

```
"template":true
```

But even if you immediately leave and go check if it is working as a template, it's still broken. So it's nothing like the page just loads with the checkbox incorrectly checked and then saves broken upon subsequent saves.

Reopening to investigate more.

**Comment by shshaw** [comment]
Definitely odd. The data being sent when saving does contain `"template": true`, and `"fork": false`, but still shows up as a fork under activity and does not save as a template. You are correct that they are not listed as forks in my profile, though. That does seem to indicate that something server-side is the culprit.

Another item to note is that I did have a Pro account during a trial period a year or two back. Weren't templates PRO-only for a little while? Perhaps an odd flag on my account & other expired-pro accounts may be the culprit.

**Comment by reccanti** [comment]
I'm not sure that would be the case in my situation. I don't think I've ever had a pro account.

**Comment by RavingAPD** [comment]
Haha I have the exact same thing going on, checked my activity and it also forks it!

**Comment by davidbullock** [comment]
I can save a maximum of 3 templates. I can only save new pens as templates if I remove pens I've previously saved. This happens every time, and doesn't seem related to whether the pen was forked or not for me. Is this deliberate for non-pro accounts?

**Comment by reccanti** [comment]
I haven't tried deleting a template, but this only started for me after I tried to add more than three templates

**Comment by shshaw** [comment]
I think that may be the key. I've never had more than three, but even
without three I still had difficulty getting them to actually save as
templates.

**Comment by yukulele** [comment]
Limited to 3 too

**Comment by rgfx** [comment]
I am limited to 3 as well.

**Comment by chriscoyier** [comment]
OK everyone!

This WILD GOOSE CHASE was ENTIRELY our fault. So very sorry. 

Our plan all along was limiting free accounts to **three templates**. Business, yadda yadda. From a back end perspective, that was working great. From a front-end perspective, we barely handled it at all. 

Once you were using all three templates, the back end just wasn't saving new ones as templates, as we intended. Now, you'll see proper UI for when you're at that limit. 

![screen_shot_2016-05-06_at_10 06 17_am](https://cloud.githubusercontent.com/assets/69156/15077269/3ab5d4e0-1373-11e6-9cb5-e97037a3af51.png)

Everyone involved in this thread, if you want to, email us at support@codepen.io and reference this thread and we'll hook you up with free PRO for a while for dealing with this one we dropped the ball on for so long.
